cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/SlowBrokerFinder.java

```diff
@@ -235,17 +235,21 @@ private void updateBrokerSlownessScore(Set<BrokerEntity> detectedMetricAnomalies
     }
     // For brokers which are previously detected as slow brokers, decrease their slowness score if their metrics has
     // recovered back to normal range.
+    Set<BrokerEntity> brokersRecovered = new HashSet<>();
     for (Map.Entry<BrokerEntity, Integer> entry : _brokerSlownessScore.entrySet()) {
       BrokerEntity broker = entry.getKey();
       if (!detectedMetricAnomalies.contains(broker)) {
         Integer score = _brokerSlownessScore.get(broker);
-        // If the score reaches zero, remove its suspicion.
         if (score != null && --score == 0) {
-          _brokerSlownessScore.remove(broker);
-          _detectedSlowBrokers.remove(broker);
+          brokersRecovered.add(broker);
```

Comment:
If this loop only uses the key of the `_brokerSlownessScore` map, why does it iterate over its entries rather than its keys?